packages/flutter/test/widgets/overlay_portal_test.dart

```diff
@@ -1365,6 +1365,191 @@ void main() {
     verifyTreeIsClean();
   });
 
+  testWidgets('PortalController can be assigned to another after deactivate', (
+    WidgetTester tester,
+  ) async {
+    final OverlayPortalController controller1 = OverlayPortalController();
+    final OverlayPortalController controller2 = OverlayPortalController();
+
+    late OverlayEntry overlayEntry1, overlayEntry2;
+
+    addTearDown(() {
+      overlayEntry1.dispose();
+      overlayEntry2.dispose();
+    });
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Overlay(
+          initialEntries: <OverlayEntry>[
+            overlayEntry1 = _buildOverlayEntry(
+              (BuildContext context) => OverlayPortal(
+                controller: controller1,
+                overlayChildBuilder: (BuildContext context) => const Placeholder(),
+              ),
+            ),
+            overlayEntry2 = _buildOverlayEntry(
+              (BuildContext context) => OverlayPortal(
+                controller: controller2,
+                overlayChildBuilder: (BuildContext context) => const Placeholder(),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Overlay(
+          initialEntries: <OverlayEntry>[
+            overlayEntry1 = _buildOverlayEntry(
+              (BuildContext context) => OverlayPortal(
+                controller: controller1,
```

Comment:
@QuncCccccc pointed out the widget tree didn't seem to have changed in the second `pump`.

Reply:
Also it looks like 4 OverlyEntries are created in this test but only 2 are disposed in the tear down process, like in the other test.

Reply:
Ohh yes, I think here same change should be done, that overlay entries should be created and then passed to `Overlay`. Let me push that change.

Reply:
@LongCatIsLooong I have made this test more simpler. Where at first there is only 1 entry which is being replaced after `pump` by another one.

Reply:
Thank you! Also I think the `attachTarget == this` check it `dispose` is probably no longer necessary?

Reply:
I have pushed the change. Actually thank you, for helping quite a lot.